src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeAssemblyBuilder.cs

```diff
@@ -37,37 +86,21 @@ public sealed partial class AssemblyBuilder : Assembly
 
         #region Constructor
 
-        internal AssemblyBuilder(AssemblyName name,
+        internal RuntimeAssemblyBuilder(AssemblyName name,
                                  AssemblyBuilderAccess access,
-                                 Assembly? callingAssembly,
-                                 AssemblyLoadContext? assemblyLoadContext,
+                                 AssemblyLoadContext assemblyLoadContext,
                                  IEnumerable<CustomAttributeBuilder>? assemblyAttributes)
         {
-            ArgumentNullException.ThrowIfNull(name);
```

Comment:
nit: add assert for `name is not null` (already validated by caller, but the assert may make it more obvious). May also apply elsewhere

Reply:
Added here only, don't think it will be useful to add Asserts for protected `XyzCore` methods.